src/thunderbots/shared/constants.h

```diff
@@ -4,6 +4,9 @@
 // and firmware code. Since this needs to be compiled by both C and C++, everything
 // should be defined in a way that's compatible with C.
 
+// define our own PI value here that is a float because M_PI in math.h is a double
+#define P_PI 3.14159265f
```

Comment:
Correct me if I'm wrong, but we redefined a `float` of PI because we thought the microcontroller would be too slow if we used `doubles` everywhere right? It was mostly done for speed?

If that is the only reason I would say just use the normal `M_PI` because it will be less confusing.

Reply:
Yeah, I think that was the rationale behind it. If it is a prohibitive performance hit though, we should keep using `float` for shared code instead of `double`.

Reply:
I agree, but I don't know if we ever determined what the impact was or if we just did it preventatively under the assumption that the impact would be severe enough to warrant using floats.

I remember someone, maybe Nick, saying a while ago that we aren't even close to using the full capability of the microcontroller, but that's not hard evidence either.

I think until we have evidence that using `double` is significantly slower, we should avoid redefining our own `float` values as it adds complexity and confusion. The microcontroller has a floating point chip, so I really don't think the overall performance difference between using `float` and `double` will be too significant.

Reply:
Changed to use double